### DISCUSSION TITLE
Unable to download S1 orbits with stackSentinel

### DISCUSSION THREAD

**Discussion by svenborgstrom** [isce-framework/isce2#406] [Q&A] [answered]
Good morning everybody,
as indicated in the subject, I'm currently unable to download S1 orbits with the "stackSentinel" script.
This is the error message I get running the script:

*****************************************
20210712
orbit was not found in the /Volumes/MacOS_Backup/LaPalmaSenDT169/orbits
downloading precise or restituted orbits ...
fetchOrbit.py -i /Volumes/MacOS_Backup/LaPalmaSenDT169/SLC/S1A_IW_SLC__1SDV_20210712T071058_20210712T071125_038741_04924E_C5D9.zip -o /Volumes/MacOS_Backup/LaPalmaSenDT169/orbits/20210712
Reference time:  2021-07-12 07:11:25
Satellite name:  S1A
Failed to find S1A orbits for tref 2021-07-12 07:11:25
Traceback (most recent call last):
  File "/Users/Sven/python/miniconda3/envs/isce2/share/isce2/topsStack/stackSentinel.py", line 787, in <module>
    main(sys.argv[1:])
  File "/Users/Sven/python/miniconda3/envs/isce2/share/isce2/topsStack/stackSentinel.py", line 750, in main
    acquisitionDates, stackReferenceDate, secondaryDates, safe_dict, updateStack = checkCurrentStatus(inps)
  File "/Users/Sven/python/miniconda3/envs/isce2/share/isce2/topsStack/stackSentinel.py", line 626, in checkCurrentStatus
    acquisitionDates, stackReferenceDate, secondaryDates, safe_dict = get_dates(inps)
  File "/Users/Sven/python/miniconda3/envs/isce2/share/isce2/topsStack/stackSentinel.py", line 283, in get_dates
    safeObj.get_orbit(inps.orbit_dirname, inps.work_dir)
  File "/Users/Sven/python/miniconda3/envs/isce2/share/isce2/topsStack/Stack.py", line 1048, in get_orbit
    self.orbit =  orbitFile[0]
IndexError: list index out of range
------------------------------------------------------------------------------------------------
The workaround I tried was to manually download all the obits, but I get another error message:

No existing stack was identified. A new stack will be generated.
*****************************************
Coregistration method:  NESD
Workflow:  interferogram
*****************************************
writing  /Volumes/MacOS_Backup/LaPalmaSenDT169/run_files/run_01_unpack_topo_reference
Traceback (most recent call last):
  File "/Users/Sven/python/miniconda3/envs/isce2/share/isce2/topsStack/stackSentinel.py", line 787, in <module>
    main(sys.argv[1:])
  File "/Users/Sven/python/miniconda3/envs/isce2/share/isce2/topsStack/stackSentinel.py", line 770, in main
    interferogramStack(inps, acquisitionDates, stackReferenceDate, secondaryDates, safe_dict, pairs, updateStack)
  File "/Users/Sven/python/miniconda3/envs/isce2/share/isce2/topsStack/stackSentinel.py", line 567, in interferogramStack
    i = slcStack(inps, acquisitionDates, stackReferenceDate, secondaryDates, safe_dict, updateStack)
  File "/Users/Sven/python/miniconda3/envs/isce2/share/isce2/topsStack/stackSentinel.py", line 453, in slcStack
    runObj.unpackStackReferenceSLC(safe_dict)
  File "/Users/Sven/python/miniconda3/envs/isce2/share/isce2/topsStack/Stack.py", line 331, in unpackStackReferenceSLC
    configObj.orbit_type = safe_dict[self.reference_date].orbitType
AttributeError: 'sentinelSLC' object has no attribute 'orbitType'
-----------------------------------------------------------------------------------------------
In this case I get the "configs" folder with an empty "config_reference" file, besides the "run_files" folder, with an empty "run_01..." file.
I would appreciate any help, thanks.
Sven.

**Comment by svenborgstrom** [comment]
Sorry,
I don't know why the text I copied from my Mac is so big, I just copied this.
I'm really sorry.
Sven.

**Reply by piyushrpt** [reply]
Using code block - https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks should take care of the font size/ formatting issues.

**Comment by piyushrpt** [comment]
This works for me. It downloaded `S1A_OPER_AUX_POEORB_OPOD_20210801T121825_V20210711T225942_20210713T005942.EOF` . 

Possible scihub outage when you tried?

**Reply by svenborgstrom** [reply]
Hi,
sorry for disturbing you again, but I'm still unable to download S1 orbits.
In particular:

- It's not a connection problem, neither from my side nor from the ESA side,  as I tried many times and in different days;
- My ISCE2 installation is in a conda environment: I removed the environment and reinstalled it again but the issue remains;
- I set another conda environment with ISCE2 inside (a different name, with other packages), in this case paying attention to properly set the path for the new environment, but the problem is always there.

Sincerely, I don't know what else can I try.
Do you think this could be a python related error?

I've also manually downloaded the orbits, but I still got the same error message pointed out in the second part of my post.

How can I try to fix those two issues?

Cheers,
Sven.

**Reply by piyushrpt** [reply]
You can add print statements in fetchOrbit.py and try to debug as to what is happening.

**Comment by BianSun** [comment]
@svenborgstrom I also met the problem😭May I ask if you have finally resolved it？And How did you specifically solve it?

**Reply by BianSun** [reply]
@svenborgstrom

**Reply by svenborgstrom** [reply, marked as answer]
Hi BianSun,
on October 31 there were some changes in the way we get the S-1 orbit files.
Everything is well explained in https://github.com/scottstanie/sentineleof
Cheers,
Sven.

**Comment by BianSun** [comment]
Thank you so much! After downloading the orbit according to this new method, the program continued to run and encountered some other error messages, but all of them were resolved later. Thanks again! Have a good day!!!👍👏

**Comment by fdelgadodelapuente** [comment]
Hi, 

You can get the precise orbits from
https://s1qc.asf.alaska.edu/aux_poeorb/

and restituted orbits from
https://s1qc.asf.alaska.edu/aux_resorb/

Francisco

**Reply by svenborgstrom** [reply]
Hi Francisco,
many thanks.
It's good to have the link also to single files.
Sven.

**Reply by Abhir-4172** [reply]
hello,
How can we download both the files from above links?

**Reply by fdelgadodelapuente** [reply]
Check on the file names which files spans the acquisition time of your S1 SLC. Then click on it.
